### ISSUE TITLE
Issues with ranges

### ISSUE THREAD

**Issue by masterkain** [WeakAuras/WeakAuras2#4836]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues.

### Description

since some days I'm experiencing issues with range checks on attackable entities. sometimes a /reload fixes the issue but it is coming up almost 100% of times now.

I attached the weakauras I'm using for testing, the issue (tested on havoc dh) is that sometimes it is very imprecise,
2 for min range and 30 for max range is being shown outside melee range, and sometimes, perhaps following a reload, it starts showing appropriate range values such as 20, 25, etc.

### WeakAuras Version

WeakAuras 5.9.2

### World of Warcraft Flavor

Retail (Default)

### World of Warcraft Region

EU

### Tested with only WeakAuras

- [X] Yes

### Reproduction Steps

```
!WA:2!1E5ZZTTXvC8WYMjjyMmXIXYjY1nLJBAQYmnQ2k2UXPPTHqM0IQuImGq)YnTIlawsSwGaqlwOF1FKk1zKvtJBNH9)aEoN4HCTD0m9(B40)c8HEQN8Faj9TlaPOSJK94orhGw8WUV9TV3N33vkZs5ImU(u3CQPZ1oNtoNd)U9Ti2R7WdcNjWlGpx2SzRL9ktD9hyfWDOC90pgEoDp2U7s4o5ndc8eSWUIa7nP8iwG)RN5ETnISjE0SwjlZLVt1MnJOIz)NF8)4l)KV6R(YUCAlCQM7esnAXdId7MmZ6SDPpFVKXjrqMmzYY3EWYJ)xl9C4pHJr8TDd4L4K20Aeo1x4Yj2c0Lr3Wisq4cnDMptOz1e)vKR2(coRvlmaV4BXthEqxhQvCZMYGGpBXk1kTyfDHmIiXCY07hfs98k7eP9GOyl6M4EuhNnB7ERntH6MRv3SGHPEmUjwHEKDO8HtRgNItZOETIvQyOSynlL4jCn8XWnsRtSFAiOLESl60Igow9nIXJs(sXEE5x2LjOhL81Kd)l1l5mxlGHoCMIlyw0Ote1RPYam5LzoHx0KWBrf5ni(TO5NXLAVoLN)2YeCAsTSp6QS9AktC1fCIGK9HJKlL5cRzwSUz157X8BgWBtKjvnl7a)MSwA9jXcCUvdvPAThI7mL7t8wkP0F)oeFwYsUbCUsWRP7h4tpIsIKBg1VLW9f67Gzx5mwtMR5ru01or7lNJC3HRP3MW8X1cxdU(la3a(jW7HdZ9Ow6kxhljo09cioL0Jq65cgTJrGuthlF2xa(zAg2EKOi5ilbYK(c5WPJzo92yPcSIBm7DSf7KZEUdFZUm0L1dI520mWl)FZaVyg4hct23grCNGT8RVflK24ObVklAUWRGhuniNgmMg8QVXBbN)lW48sH5IJORrSIc4QA)8bo0gWfHVZdo2EIT89gPybxcJrzfeMqdUS2(YjljSgWBaFpyC4cW3xd3IdudXTbk1iXHcb2zsS8qhoxd42OBdp)I4cr3sKnguolsWSJWLug9b2)GnlBsntWqzK85phm99F46uAybzMtyiRrUWBdtITjreHSOrD7ImUHQ1nAomc6JVkfmKTW(ohoEyobDBXArUembTsc3MjCcLr1JeIATO0bW16m8B9(bxDQ2mFv2i8CJ4NJ1bEqI1bmVEHfnRg(QkJixgibp7Lzocx9c4RpmzhzBtDug3RFcOVAACDK673ngtmn3bpM8abEenrJ6luDHIWRXXtNCoDvt0EGGy2oPhfFr45kXz7M)JIjoYwP8MMNidmyN6RSTf2aUmNeU)YPdscGnzrmzHl9mpstE4yLxyHIgRPx1e7hnkF7zndV8JLlpoRnmP2Fyak1tNi9CVYjIg5xvkFvx0Ss5fkE44GrgOo2IzALOIalPs6WYWkWQWDGF1EWhNb(1WKWVbwdPVgarLoalWgCYauOj0Qb42pjQnRwRsXsMWDrxU(ldEzG2G))PFlVGTkXPBet9T3P23(ktn91tY0YpCecZ32lTK7UV0Kmg7Bfleb(vXlyqL2UsZvuIjJ3toEqAUtRblvfxhjF9wPInzvZCykq5zxM96(0OOSQfwxDJLUCORYqfg(XXu7MEY1ywUuwlxrb4Nwc(qPmc8Zvp)fQNFG65lPlvrGFuw4hRbVpQ(mmEG9AOVBqq7mqHElV(88sRSQ5QRBbpV8cZtx9(DYppZhUQMsszMuThSJfBqXU93XQ(mgflUGXwseVaCfnoZxErm11G4f6sYgE(bcws0(wSiPmHJBphwK8ERYy0(6VYxS0(dMvd4DvA3OsRuWDG(7nFml5Eulh(MWmOW5aDt4wnGINMajQCjLdvsvWSkzjLKNu(6u16GF5q5nOsQugmVs5cwqPvbvDHAsLj4Jo8cdbAL0czBvE9mP6V(2R0voS96myFGLW7tKY7J0vbbpRnuq4J0ebBmOXb4qKliGyyZXHTYaBN4GDYc7Mb(TzHFxw437c)HXGpXf(JNn3c7neAH9haSWFssRRCR5kTT)gV3DlpZtbTs2gPv4aKnH7bhkrs4p7cFAw4V4cFMe1G7))lJ9IF23AecddZthSmAgyhh9ucxxc6mGUWPEc46znGhLihbgw8zggoDcZOK6O(StzO7sPmCl(gJYqsa5JeqBuXHuOdPKc9Q9U8YnF7RUAZI8JjUjuNVVw5r5FTgsVNG4oqrVdjouDibEWn)K8tsqmMkoodfQoPqKKGq64PsHQZtPcf8xp7Y2F7jP6OkDtKw6gf0(7FZO6G1(hJhqgb5bKeqEaPIr5Hp4X4HpCipCWa1Gh5V4)Mp5)hazDnPaJ36Cm)G1)Ni0GQuF6qGbVqEYiHoX)oxJFfj7GSeYoVVct3RrUn)3R8)o
```

4 weakauras for testing minrange maxrange of both target and focus

### Screenshots

so here we have the same targeted mob and my char is in the same position for this test.

- the first image is about targeting the mob right after login (2/30, bugged out)
- the second image is targeting the mob right after a /reload (15/20, correct, but sometimes not even after a reload it will fix itself)

![Screenshot 2024-01-26 061742](https://github.com/WeakAuras/WeakAuras2/assets/12844/760745df-1844-49ea-9e0b-70f96906eded)
![Screenshot 2024-01-26 061601](https://github.com/WeakAuras/WeakAuras2/assets/12844/4a281f33-5918-4b82-a57f-26ac20e3a397)

**Comment by masterkain** [comment]
so we have the same mob, my char is the same position
the first image is targeting right after login
the second image is targeting right after a /reload

![Screenshot 2024-01-26 061742](https://github.com/WeakAuras/WeakAuras2/assets/12844/760745df-1844-49ea-9e0b-70f96906eded)
![Screenshot 2024-01-26 061601](https://github.com/WeakAuras/WeakAuras2/assets/12844/4a281f33-5918-4b82-a57f-26ac20e3a397)

**Comment by InfusOnWoW** [comment]
This is a bug in the LibRangeCheck-3.0 library, which I just fixed. It should work better in the next release.

**Comment by masterkain** [comment]
hello, I tested for few days LibRangeCheck 3.0 MINOR_VERSION 11 and I'm afraid the issue still persists.

**Comment by InfusOnWoW** [comment]
I just tested your WA again and it correctly works even after a fresh game start. I suspect you didn't correctly install the new library.

**Comment by masterkain** [comment]
what I'm finding is that it isn't 100% reproducible every time. it even happened in combat at some point. regarding the install I just took the updated lua file and placed into the weakaura libs directory.

if is there's any test I can do please let me know.

**Comment by InfusOnWoW** [comment]
That's not the same issue description. I obviously tested exactly the case you described in your original description, because you wrongly claimed that "the issue persist".

If you know how to reproduce your new issue, please properly describe how to reproduce it.

**Comment by masterkain** [comment]
maybe there's a misunderstanding, the original issue is still what I'm experiencing -- try maybe keeping the weakaura enabled for few hours and give it an eye, if you see 8-30 or some big range like that (instead of granular 20, 25, etc.) that's the issue, tested with DH and DK at the moment, not sure if it matters.

on second thought, regarding the in combat portion of my previous comment, disregard that because I need to further review some vods to be sure.

thanks for looking into it

**Comment by masterkain** [comment]
this is still very much an issue, maybe it has something to do with the load order of the addon requiring librangecheck, I don't know.

but it's 100% reproducible on a warrior, when you zone into an instance range values will not be correct, only after a reload we get a chance to get them working properly.
